istio/istio.go

```diff
@@ -503,9 +503,10 @@ func (iClient *IstioClient) ApplyOperation(ctx context.Context, arReq *meshes.Ap
 			return nil, err
 		}
 	case installSMI:
-		if !arReq.DeleteOp && arReq.Namespace != "default" {
-			iClient.createNamespace(ctx, arReq.Namespace)
-		}
+		// Always set the namespace for SMI adapter to istio-system
+		// as this is where istio is installed and how clusterrolebinding 
+		// for the SMI adapter is defined
+		arReq.Namespace = "istio-system"
```

Comment:
At first blush, this looks good, @antweiss. As you noodle on it, what you arrive at is the inevitable notion that we'll want to allow users to identify a namespace for control plane deployment other than the default "Istio-system".  

While Istio supports this, Meshery is not incorporating a user-defined namespace in it's provisioning of Istio. Given that SMI is a young project with little capability thus far, we've not included it by default in the Istio deployment. The ideally solution here is for user-defined namespaces to accounted for when deploying Istio and for an "enable SMI" to be a checkbox be presented as a choice (later to be enabled by default) when deploying Istio. The same goes for "enable mTLS" as a checkbox choice. The UI (and the installation procedure behind it) are in need of a slightly broader uplift (that if you're inclined to undertake, that would be wonderful).

@girishranganathan do you have additional thoughts here?

@antweiss we may want to break out the Dockerfile update, so that it might land despite the need for a more holistic approach here. I'm stoked that you've jumped in like this. This is wonderful.

Reply:
Thanks @leecalcote - this makes total sense.
It would be a much larger undertaking but I agree it's the right thing to do.
As I see it - this can be broken down into phases:
1. Allow installing Istio into a user-defined namespace
2. Allow installing SMI adapter into a user-defined namespace
3. Convert SMI and mTLS into checkbox options in the UI

Consequently I'll do the following:
1. Break out the Dockerfile fix into a separate PR
2. Close https://github.com/layer5io/meshery-istio/issues/25
3. Open new issues for each of the phases. 
4. Start working on them one by one.

Any comments welcome.

Reply:
Opened https://github.com/layer5io/meshery-istio/pull/28

Reply:
after looking into it - I decided not to close https://github.com/layer5io/meshery-istio/issues/25 . Instead I edited it to reflect the new course of action.

Reply:
Opened https://github.com/layer5io/meshery/issues/337 for the UI change.
Closing this PR now